packages/libs/core/src/locale.ts

```diff
@@ -0,0 +1,57 @@
+import Accept from "@hapi/accept";
+import { Manifest, RoutesManifest } from "./types";
+
+export function addDefaultLocaleToPath(
+  path: string,
+  routesManifest: RoutesManifest
+): string {
+  if (routesManifest.i18n) {
+    const defaultLocale = routesManifest.i18n.defaultLocale;
+    const locales = routesManifest.i18n.locales;
+    const basePath = routesManifest.basePath;
+
+    // If prefixed with a locale, return that path
+    for (const locale of locales) {
+      if (
+        path === `${routesManifest.basePath}/${locale}` ||
+        path.startsWith(`${routesManifest.basePath}/${locale}/`)
+      ) {
+        return path;
+      }
+    }
+
+    // Otherwise, prefix with default locale
+    if (path === "/" || path === `${basePath}`) {
+      return `${basePath}/${defaultLocale}`;
+    } else {
+      return path.replace(`${basePath}/`, `${basePath}/${defaultLocale}/`);
+    }
+  }
+
+  return path;
+}
+
+export const getAcceptLanguageLocale = (
+  acceptLanguage: string,
+  manifest: Manifest,
+  routesManifest: RoutesManifest
+) => {
+  if (routesManifest.i18n) {
+    const locales = routesManifest.i18n.locales;
+    const defaultLocale = routesManifest.i18n.defaultLocale;
+
+    const preferredLanguage = Accept.language(acceptLanguage);
+
+    // Find language in locale that matches preferred language
+    for (const locale of locales) {
+      if (preferredLanguage.toLowerCase() === locale.toLowerCase()) {
```

Comment:
I know I wrote this code before but maybe you can to `toLowerCase()` in line 43 instead to help with perf a bit

Reply:
I would hope that V8 would to that anyway, but might as well.

However, I meant to ask about that code. Is there any reason it uses `Accept.language(header)` and matches manually, rather than letting the library do the matching: https://hapi.dev/module/accept/api/?v=5.0.2#languagelanguageheader-preferences

I think that would allow a better change of matching the user's preferences. Totally orthogonal to this PR though.

Reply:
Yea, I think there was no reason - I might have just missed that code. Feel free to update it!